### ISSUE TITLE
Hide comments section when no comments

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#2926]
If there are no comments for a node, and the comments have been turned off for that content type, the comments section is still displayed in the HTML (albeit empty). It shouldn't be displayed when there are no comments and the comment form is disabled.

Here's what it looks like currently:
```html
<article id="node-2" class="node node-page view-mode-full clearfix" role="article">
  <div class="content">
    ...
  </div>
  <section class="comments" id="comments">
  </section>
</article>
```
If you've styled the comments section to have a background color, border, etc., then that shows, even though there are no comments/comment form...

---

PR by @BWPanda: https://github.com/backdrop/backdrop/pull/2048

**Comment by olafgrabienski** [comment]
@BWPanda Thanks for the PR! Makes perfect sense in my opinion, and works as expected. Code looks super simple but I'm not really able to evaluate it.

**Comment by olafgrabienski** [comment]
@BWPanda Btw, I saw the PR only by accident. Better write a short comment when you make a PR so that followers get notified of it.

**Comment by quicksketch** [comment]
Thanks @BWPanda for finding and filing this issue! I looked at the PR and it seems like it's causing a number of PHP notices. Apparently the entire `$comments` variable may be missing in some situations, causing undefined variable notices. Could you reroll?

**Comment by BWPanda** [comment]
No problem, just added another commit.

@olafgrabienski Not sure what you're referring to... Where should I write comments?

**Comment by olafgrabienski** [comment]
> Where should I write comments?

In the issue itself. When I follow an issue, I get notifications about comments in the issue via email but I don't get a notification about a new PR (unless someone mentions the PR in an issue comment).

**Comment by BWPanda** [comment]
Oh right. Ok, thanks. I'll do that next time!

**Comment by jenlampton** [comment]
This PR looks great and works for me too, but I'd love to see this improvement to Bartik as well as core.

**Comment by BWPanda** [comment]
Updated PR with @jenlampton's suggestion.

**Comment by jenlampton** [comment]
Looks great!  RTBC from me :)

**Comment by quicksketch** [comment]
I tested out the PR at https://github.com/backdrop/backdrop/pull/2048, but it didn't seem to actually work. The `isset()` checks actually pass, because `$comments['comment_form']` is set to an empty array, even when there is no comment form. So the wrapper still prints out just as before.

We could fix this by using the ugly, but functional condition `if (!empty($comments['comment_form']))` but we should give the template better variables to work with so it doesn't have to do back-flips like this. I filed a PR at https://github.com/backdrop/backdrop/pull/2160 that reworks the variables that are passed to the node template. Now `$comments` is simply `FALSE` if comments are not shown. And no other conditions are necessary.

**Comment by jenlampton** [comment]
I love the new solution! 1. [it works](http://2160.backdrop.backdrop.qa.backdropcms.org/posts/post-no-comments-or-form) and 2. the template file is much cleaner.

**Comment by quicksketch** [comment]
It does seem to work... but it's failing tests. Sad trombone. Looks like failures are directly related to comment changes. This will need an updated commit or replacement PR.

**Comment by quicksketch** [comment]
The problem here was caused by comments being moved later in the page request. Since the comments were rendered later, the node was getting marked as viewed before they were rendered. This caused the "new" indicator to go missing from comments. I updated the PR at https://github.com/backdrop/backdrop/pull/2160 to move updating the "history" table to the end of the request, which should actually net an insignificant performance improvement, since Backdrop will execute shutdown functions after the page has been returned to the user, removing a DB write from the page building process.
